File: app/src/main/java/pt/estga/spotme/ui/home/HomeFragment.java
package pt.estga.spotme.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;

import pt.estga.spotme.R;
import pt.estga.spotme.database.AppDatabase;
import pt.estga.spotme.databinding.FragmentHomeBinding;
import pt.estga.spotme.entities.Parking;
import pt.estga.spotme.utils.UserSession;

import java.util.concurrent.Executors;

public class HomeFragment extends Fragment {

    private FragmentHomeBinding binding;
    private AppBarConfiguration appBarConfiguration;

    private UserSession userSession;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        HomeViewModel homeViewModel = new ViewModelProvider(this).get(HomeViewModel.class);
        userSession = UserSession.getInstance(requireContext());

        binding = FragmentHomeBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        NavController navController = Navigation.findNavController((AppCompatActivity) getActivity(), R.id.nav_host_fragment_content_main);
        setupSetMyCarImageClickListener(root, navController);
        setupFindMyCarImageClickListener(root, navController);

        return root;
    }

    private void setupSetMyCarImageClickListener(View root, NavController navController) {
        ImageView setMyCarImage = root.findViewById(R.id.setMyCarImage);
        if (setMyCarImage != null) {
            setMyCarImage.setOnClickListener(v -> navController.navigate(R.id.parkingFormFragment));
        }
    }

    private void setupFindMyCarImageClickListener(View root, NavController navController) {
        ImageView findMyCarImage = root.findViewById(R.id.pickMyCarImage);
        if (findMyCarImage != null) {
            findMyCarImage.setOnClickListener(v -> {
                AppDatabase db = AppDatabase.getInstance(requireContext());
                Executors.newSingleThreadExecutor().execute(() -> {
                    Parking lastParking = db.parkingDao().getLastParkingByUserId(userSession.getUserId());
                    if (lastParking != null) {
                        Bundle bundle = new Bundle();
                        bundle.putSerializable("parking", lastParking);
                        requireActivity().runOnUiThread(() -> navController.navigate(R.id.parkingDetailViewFragment, bundle));
                    } else {
                        requireActivity().runOnUiThread(() -> {
                            Toast.makeText(requireContext(), "No parking records found", Toast.LENGTH_SHORT).show();
                        });
                    }
                });
            });
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/pt/estga/spotme/ui/home/HomeFragment.kt
package pt.estga.spotme.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import pt.estga.spotme.ui.BaseFragment
import pt.estga.spotme.databinding.FragmentHomeBinding
import pt.estga.spotme.navigation.HomeNavigationManager

class HomeFragment : BaseFragment() {

    private var _binding: FragmentHomeBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentHomeBinding.inflate(inflater, container, false)
        setupNavigationListeners()
        return binding.root
    }

    private fun setupNavigationListeners() {
        val navController = findNavController()

        val setLocationClickListener = View.OnClickListener {
            navController.navigate(pt.estga.spotme.R.id.parkingFormFragment)
        }

        binding.setLocationCard.setOnClickListener(setLocationClickListener)
        binding.setMyCarImage.setOnClickListener(setLocationClickListener)
        binding.setLocationButton.setOnClickListener(setLocationClickListener)

        val findCarClickListener = View.OnClickListener {
            val userId = userViewModel.userId.value
            if (userId != null) {
                HomeNavigationManager.navigateToLastParking(
                    context = requireContext(),
                    navController = navController,
                    userId = userId,
                    lifecycleScope = viewLifecycleOwner.lifecycleScope
                ) {
                    Toast.makeText(requireContext(), "No parking records found", Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(requireContext(), "Utilizador não encontrado.", Toast.LENGTH_SHORT).show()
            }
        }

        binding.findLocationCard.setOnClickListener(findCarClickListener)
        binding.pickMyCarImage.setOnClickListener(findCarClickListener)
        binding.findMyCarButton.setOnClickListener(findCarClickListener)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}